File: app/src/main/java/co/tpcreative/supersafe/common/services/SuperSafeReceiver.java
package co.tpcreative.supersafe.common.services;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

public class SuperSafeReceiver extends BroadcastReceiver {

    public static final String TAG = SuperSafeReceiver.class.getSimpleName();
    public static ConnectivityReceiverListener connectivityReceiverListener;

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG,"onReceive");
        String action=intent.getAction();
        /*In the case status of network changed and then updating status for app(Connected/Disconnect)*/
        if (action.equalsIgnoreCase(ConnectivityManager.CONNECTIVITY_ACTION)) {
            ConnectivityManager cm = (ConnectivityManager) context
                    .getSystemService(Context.CONNECTIVITY_SERVICE);
            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
            boolean isConnected = activeNetwork != null
                    && activeNetwork.isConnected();
            if (connectivityReceiverListener != null) {
                connectivityReceiverListener.onNetworkConnectionChanged(isConnected);
            }
        }
    }

    public static boolean isConnected() {
        ConnectivityManager
                cm = (ConnectivityManager) SuperSafeApplication.getInstance().getApplicationContext()
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null
                && activeNetwork.isConnectedOrConnecting();
    }

    public interface ConnectivityReceiverListener {
        void onNetworkConnectionChanged(boolean isConnected);
    }


}

File: app/src/main/java/co/tpcreative/supersafe/common/services/SuperSafeReceiver.kt
package co.tpcreative.supersafe.common.services
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import co.tpcreative.supersafe.common.util.NetworkUtil

class SuperSafeReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent) {
        val action: String? = intent.action
        /*In the case status of network changed and then updating status for app(Connected/Disconnect)*/
        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION, ignoreCase = true)) {
            connectivityReceiverListener?.onNetworkConnectionChanged(!NetworkUtil.pingIpAddress(SuperSafeApplication.getInstance()))
        }
        if (action.equals(Intent.ACTION_SCREEN_OFF, ignoreCase = true)) {
            connectivityReceiverListener?.onActionScreenOff()
        }
    }

    interface ConnectivityReceiverListener {
        fun onNetworkConnectionChanged(isConnected: Boolean)
        fun onActionScreenOff()
    }

    companion object {
        val TAG = SuperSafeReceiver::class.java.simpleName
        var connectivityReceiverListener: ConnectivityReceiverListener? = null
        fun isConnected(): Boolean {
            return !NetworkUtil.pingIpAddress(SuperSafeApplication.getInstance())
        }
    }
}